### ISSUE TITLE
Python 3.10 RC1 missing in pyenv

### ISSUE THREAD

**Issue by memark** [pyenv/pyenv#2036]
Too many issues will kill our team's development velocity, drastically.
Make sure you have checked all steps below.

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): MacOS 11.5.1
- [x] OS architecture (e.g. amd64): x86_64
- [x] pyenv version: 2.0.4

Python 3.10 RC1 has been released on 2021-08-02.
https://www.python.org/downloads/release/python-3100rc1/

However in pyenv I can still only find `3.10.0b4` and `3.10-dev`:
```
# pyenv install --list | grep 3.10
    3.10.0b4
    3.10-dev
    miniconda-3.10.1
    miniconda3-3.10.1
```

**Comment by native-api** [comment]
For prereleases, we only keep the latest one, overwriting it when a new one is released.

**Comment by native-api** [comment]
Oh, sorry, RC1 _is_ the latest one.

[It is present atm](https://github.com/pyenv/pyenv/blob/8b60418361e29d30fbe7ee3133e57f4d351464da/plugins/python-build/share/python-build/3.10.0rc1), please update Pyenv to the latesty HEAD.

**Comment by anton-petrov** [comment]
@memark you can find python 3.10rc1 in in the just released pyenv 2.0.5

**Comment by memark** [comment]
>Oh, sorry, RC1 is the latest one.

Exactly. :)

>you can find python 3.10rc1 in the just released pyenv 2.0.5

Thank you!
